### DISCUSSION TITLE
Using Git Actions secrets in a forked repository

### DISCUSSION THREAD

**Discussion by silvzr** [community/community#140642] [Actions]
### Select Topic Area

Question

### Body

Hi, I've forked a repository with the intent of modifying the code for personal use (different interests with the original repository owner) instead of submitting PRs. I spent some hours trying to understand why secrets didn't work and I've just found out that they don't on forked repositories. I understand that for security concerns I shouldn't be able to access secrets of the original repo owner but why can't I use the secrets I've created? I know I could submit a PR which could make show their secrets but this is also disabled by default (rightfully).
This doesn't really make sense to me as this forces me and everyone who want to use my repo to clone the repository instead of forking, making the merge of new commits way less practical than with a fork. 
Maybe I'm missing something thanks to which they will work but I've already spent hours looking into it or checking if my workflow was coded uncorrectly, I hope someone could help me with this (if a solution exists).
Thanks in advance

**Comment by FabDonRixos** [comment]
To use secrets in a forked repository without cloning, you'll need to enable workflows to run from the fork. Here's how to do it:

    Go to the repository settings in your fork.
    Enable GitHub Actions in the "Actions" tab.
    Create the secrets in your fork (under "Settings" -> "Secrets and variables" -> "Actions").

However, note that workflows triggered by Pull Requests from forks to the upstream repository still won't have access to secrets for security reasons. If you are only using the fork for personal purposes, and not creating PRs, this method should work for running your own workflows.

I Hope I undestood you question right and maybe this will help 🤷‍♂️.

**Reply by silvzr** [reply]
Hello, first of all thanks for your answer. I've already enabled actions and set my personal secrets in the repository settings, but when I launch the workflow which uses them they get displayed in logs as blank and not as *** (this means the secret wasn't found). I spent a lot of time checking if my code was correct and I'm pretty sure it is, also cause I took it from another repo which is able to run it correctly (not a fork).

`          if [ ! -z "${{ secrets.KEYSTORE }}" ]; then
            {
              echo KEYSTORE_PASSWORD='${{ secrets.KEYSTORE_PASSWORD }}'
              echo KEY_ALIAS='${{ secrets.KEY_ALIAS }}'
              echo KEY_PASSWORD='${{ secrets.KEY_PASSWORD }}'
              echo KEYSTORE_FILE='key.jks'
            } >> gradle.properties
            echo ${{ secrets.KEYSTORE }} | base64 -d > key.jks
          fi`
          
this is the code, I've called my secrets KEYSTORE, KEY_ALIAS, KEY_PASSWORD and KEYSTORE_PASSWORD. They're repository secrets so they're not specific to an environment. Is there something I'm doing wrong?

**Reply by FabDonRixos** [reply]
Hi again, maybe you could do it like this

```
if [[ -n "${{ secrets.KEYSTORE }}" ]]; then
  echo "KEYSTORE_PASSWORD='${{ secrets.KEYSTORE_PASSWORD }}'" >> gradle.properties
  echo "KEY_ALIAS='${{ secrets.KEY_ALIAS }}'" >> gradle.properties
  echo "KEY_PASSWORD='${{ secrets.KEY_PASSWORD }}'" >> gradle.properties
  echo "KEYSTORE_FILE='key.jks'" >> gradle.properties
  echo "${{ secrets.KEYSTORE }}" | base64 -d > key.jks
fi
```
or
```
if [[ ! -z "${{ secrets.KEYSTORE }}" ]]; then
  {
    echo "KEYSTORE_PASSWORD='${{ secrets.KEYSTORE_PASSWORD }}'"
    echo "KEY_ALIAS='${{ secrets.KEY_ALIAS }}'"
    echo "KEY_PASSWORD='${{ secrets.KEY_PASSWORD }}'"
    echo "KEYSTORE_FILE='key.jks'"
  } >> gradle.properties
  echo "${{ secrets.KEYSTORE }}" | base64 -d > key.jks
fi
```

you could also try writing them into a env, I remember that this helped me once maby it will work in your scenario too.
```
env:
   KEY_PASSWORD: ${{ secrets.KEY_PASSWORD }}
```
And please ensure you really set your Secrets correctly
And go through your settings and check if there are some settings that sound like they need to be set diffrently like this one:

![grafik](https://github.com/user-attachments/assets/d01ced40-7d41-4bce-9408-e79af569c9da)

Happy testing 🍀

**Reply by silvzr** [reply]
I've found out the reason: the code in this workflow was correct but I had to add "secrets: inherit" to the job that used it in the workflow from which the Action starts

**Reply by FabDonRixos** [reply]
ohh.

Great to hear you did it.
Have a nice rest of your weekend.
